src/transformStateWithClones.js

```diff
@@ -6,8 +6,66 @@
  *
  * @return {Object[]}
  */
+/* function transformStateWithClones(state, actions) {
+  let cloneState = { ...state };
+  const result = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        if (action.extraData) {
+          cloneState = { ...cloneState, ...action.extraData };
+        }
+        break;
+
+      case 'removeProperties':
+        if (Array.isArray(action.keysToRemove)) {
+          cloneState = { ...cloneState };
+
+          for (const key of action.keysToRemove) {
+            delete cloneState[key];
+          }
+        }
+        break;
+
+      case 'clear':
+        for (const key in cloneState) {
+          cloneState = {};
+        }
+        break;
+    }
+  }
+  result.push({ ...cloneState });
+
+  return result;
+} */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let cloneState = { ...state };
+  const result = [];
+
+  for (const i of actions) {
+    switch (i.type) {
+      case 'addProperties':
+        Object.assign(cloneState, i.extraData);
```

Comment:
Ensure that `i.extraData` is not undefined before using `Object.assign`. This will prevent potential errors if `extraData` is not provided in the action.